File: bots/server_module/game_state/order_type.py
from enum import StrEnum, auto


class OrderType(StrEnum):
    MARCH = auto()
    DEFEND = auto()
    SUPPORT = auto()
    RAID = auto()
    CONSOLIDATE_POWER = 'consolidatePower'

    @classmethod
    def from_str(cls, string: str):
        if string == 'consolidatePower':
            return cls.CONSOLIDATE_POWER
        else:
            return cls[string.upper()]

File: game-logic-core/scala3MainProject/src/main/scala/fwc/game/planningPhase/OrderType.scala
package fwc.game.planningPhase

sealed trait OrderType

object OrderType {
  case object March extends OrderType {
    override def toString = "march"
  }

  case object Defend extends OrderType {
    override def toString = "defend"
  }

  case object Support extends OrderType {
    override def toString = "support"
  }

  case object Raid extends OrderType {
    override def toString = "raid"
  }

  case object ConsolidatePower extends OrderType {
    override def toString = "consolidatePower"
  }

  def fromString(str: String): OrderType = {
    str match {
      case "march" => March
      case "defend" => Defend
      case "support" => Support
      case "raid" => Raid
      case "consolidatePower" => ConsolidatePower
      case _ => throw new RuntimeException(s"Unknown order type $str")
    }
  }
}
